src/pages/AssignServerRoles.js

```diff
@@ -336,6 +351,22 @@ class AssignServerRoles extends BaseWizardPage {
       };});
     }
 
+    // if it is update and imported server has the same id or ip-addr or mac-addr or ilo-ip
+    // as any server in the model already, it will be ignored
+    if(this.props.mode && this.props.mode === 'addserver') {
+      servers = servers.filter(server => {
+        const found = model.getIn(['inputModel', 'servers']).some(svr => {
+          return (
+            svr.get('id') === server['id'] ||
+            (svr.get('ip-addr') && svr.get('ip-addr') === server['ip-addr']) ||
```

Comment:
These checks before the && are also unnecessary. It should just be `svr.get('ip-addr') === server['ip-addr']` and so on

Reply:
ip-addr might be ok to remove since import csv logic already make sure it is not empty. However for mac-addr, ilo-ip, could be empty to start with if user discovered something,  if both of them empty, it could be treated as equal.  
svr.get('ilo-ip') is undefined server['ilo-ip'] is undefined...svr.get('ilo-ip') === server['ilo-ip'] is true

Reply:
will update comments